eth/backend.go

```diff
@@ -186,10 +186,19 @@ func New(ctx *node.ServiceContext, config *Config) (*Ethereum, error) {
 	}
 
 	// Quorum: Set protocol Name/Version
+	// keep `var protocolName = "eth"` as is, and only update the quorum consensus specific protocol
+	// This is used to enable the eth service to return multiple devp2p subprotocols.
+	// Previously, for istanbul/64 istnbul/99 and clique (v2.6) `protocolName` would be overridden and
+	// set to the consensus subprotocol name instead of "eth", meaning the node would no longer
+	// communicate over the "eth" subprotocol, e.g. "eth" or "istanbul/99" but not eth" and "istanbul/99".
+	// With this change, support is added so that the "eth" subprotocol remains and optionally a consensus subprotocol
+	// can be added allowing the node to communicate over "eth" and an optional consensus subprotocol, e.g. "eth" and "istanbul/100"
 	if chainConfig.IsQuorum {
 		quorumProtocol := eth.engine.Protocol()
-		protocolName = quorumProtocol.Name
-		ProtocolVersions = quorumProtocol.Versions
+		// set the quorum specific consensus devp2p subprotocol, eth subprotocol remains set to protocolName as in upstream geth.
+		quorumConsensusProtocolName = quorumProtocol.Name
```

Comment:
We are mirroring the way the eth `protocolName` `ProtocolVersions` is set for consistency:
eth/backend.go
```
func (s *Ethereum) Protocols() []p2p.Protocol {
	protos := make([]p2p.Protocol, len(ProtocolVersions))
	for i, vsn := range ProtocolVersions {
```	
```
func (s *Ethereum) quorumConsensusProtocols() []p2p.Protocol {
	protos := make([]p2p.Protocol, len(quorumConsensusProtocolVersions))
	for i, vsn := range quorumConsensusProtocolVersions {
```

Reply:
Ok I get the point.

Still I see a difference

```go
func (s *Ethereum) Protocols() []p2p.Protocol {
	protos := make([]p2p.Protocol, len(ProtocolVersions))
	for i, vsn := range ProtocolVersions {
```

uses `ProtocolVersions` global variable which is constant while

```go
func (s *Ethereum) quorumConsensusProtocols() []p2p.Protocol {
	protos := make([]p2p.Protocol, len(quorumConsensusProtocolVersions))
	for i, vsn := range quorumConsensusProtocolVersions {
```

uses `quorumConsensusProtocolVersions` global variable which is not constant. 

`quorumConsensusProtocolVersions` is indirectly set by `New(...) (*Ethereum, error)` which I think is not a good practice and maybe error-prone (e.g it creates coupling between `Ethereum` structs through a global variable which is not constant).

I think it would be better to have a `quorumConsensusProtocolVersions` attribute on the `Ethereum` struct that is set on `New(...)` then `quorumConsensusProtocols()` could re-use the attribute or something like this.

What do you think?

Reply:
Before this change, the `ProtocolVersions` and `ProtocolName` vars would be overridden here to set them to the  quorum specific subprototol obtained from the consensus engine.  In upstream (same geth version), they would not be overridden. This change leaves `ProtocolName` `ProtocolVersions` always set to the `eth` subprotocol. 
With this initial dual subprotocol approach, keeping the changes as minimally invasive to the existing code as possible was the approach. In the (near) future, this will be refactored more (consensus may be a service with its own protocol, when we get up to date with the current geth, eth service will be refactored to support more than one subprotocol, e.g. `snap`, `eth`, `quorumconsensus`. To the client, they won't see a difference once they are running the dual subprotocols, but the Quorum code can refactored to support a different subprotocol design.

Reply:
Ok